app/src/main/kotlin/co/netguru/baby/monitor/client/feature/settings/ConfigurationViewModel.kt

```diff
@@ -14,28 +17,63 @@ import javax.inject.Inject
 
 class ConfigurationViewModel @Inject constructor(
     private val resetAppUseCase: ResetAppUseCase,
-    private val firebaseRepository: FirebaseRepository
+    private val firebaseRepository: FirebaseRepository,
+    private val voiceAnalysisUseCase: VoiceAnalysisUseCase
 ) : ViewModel() {
 
     private val compositeDisposable = CompositeDisposable()
-    private val mutableResetState = MutableLiveData<ResetState>()
-    val resetState: LiveData<ResetState> = mutableResetState
+    private val mutableResetState = MutableLiveData<ChangeState>()
+    val resetState: LiveData<ChangeState> = mutableResetState
 
-    fun resetApp(messageSender: MessageSender? = null) {
-        resetAppUseCase.resetApp(messageSender)
-            .doOnSubscribe { mutableResetState.postValue(ResetState.InProgress) }
+    private val mutableVoiceAnalysisOptionState =
+        MutableLiveData<Pair<ChangeState, VoiceAnalysisOption?>>()
+    val voiceAnalysisOptionState: LiveData<Pair<ChangeState, VoiceAnalysisOption?>> =
+        mutableVoiceAnalysisOptionState
+
+    fun resetApp(messageController: MessageController? = null) {
+        resetAppUseCase.resetApp(messageController)
+            .doOnSubscribe { mutableResetState.postValue(ChangeState.InProgress) }
             .subscribeOn(Schedulers.io())
             .subscribeBy(
                 onComplete = {
-                    mutableResetState.postValue(ResetState.Completed)
+                    mutableResetState.postValue(ChangeState.Completed)
                 },
                 onError = {
-                    mutableResetState.postValue(ResetState.Failed)
+                    mutableResetState.postValue(ChangeState.Failed)
                     Timber.w(it)
                 }
             ).addTo(compositeDisposable)
     }
 
+    fun chooseVoiceAnalysisOption(
+        messageController: MessageController,
+        voiceAnalysisOption: VoiceAnalysisOption
+    ) {
+        voiceAnalysisUseCase.chooseVoiceAnalysisOption(
+            messageController,
+            voiceAnalysisOption
+        )
+            .doOnSubscribe { mutableVoiceAnalysisOptionState.postValue(ChangeState.InProgress to null) }
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeBy(
+                onSuccess = { success ->
+                    mutableVoiceAnalysisOptionState.postValue(
+                        if (success) ChangeState.Completed to voiceAnalysisOption
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/8b49d288c8cda524f66973edfcfe165e7a30812e/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/settings/ConfigurationViewModel.kt#L62"></span>Multi-line if statement was found that does not have braces. These should be added to improve readability.</td>
    </tr>
  </tbody>
</table>